client/components/scheduled-updates/scheduled-updates-gate/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { useTranslate } from 'i18n-calypso';
+import { FC, ReactNode, FocusEvent, useState } from 'react';
+import Notice from 'calypso/components/notice';
+import NoticeAction from 'calypso/components/notice/notice-action';
+import HostingActivateStatus from 'calypso/my-sites/hosting/hosting-activate-status';
+import { useDispatch, useSelector } from 'calypso/state';
+import { transferInProgress } from 'calypso/state/automated-transfer/constants';
+import { getAutomatedTransferStatus } from 'calypso/state/automated-transfer/selectors';
+import { initiateThemeTransfer } from 'calypso/state/themes/actions';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import UpsellNudgeNotice from './upsell-nudge';
+
+import './style.scss';
+
+interface ScheduledUpdatesGateProps {
+	hasScheduledUpdate: boolean;
+	isAtomic: boolean;
+	children: ReactNode;
+}
+
+const ScheduledUpdatesGate: FC< ScheduledUpdatesGateProps > = ( {
+	hasScheduledUpdate,
+	isAtomic,
+	children,
+} ) => {
+	const translate = useTranslate();
+	const siteId = useSelector( getSelectedSiteId );
+	const dispatch = useDispatch();
+	const transferState = useSelector( ( state ) => getAutomatedTransferStatus( state, siteId ) );
+	const [ hasTransfer, setHasTransferring ] = useState(
+		!! (
+			transferState &&
+			transferInProgress.includes( transferState as ( typeof transferInProgress )[ number ] )
+		)
+	);
```

Comment:
I'm not exactly sure what the current best practice is around passing down props vs getting them from state within a component. Like, why are we getting the selected site id from state but pass down `hasScheduledUpdate`. Just food for thought (and maybe research? or feedback from calypso team? I'd love to know, too! :) )